File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> num_idx;
        for(size_t i = 0; i < nums.size(); i++) {
            if(num_idx.contains(target - nums[i])) {
                return {num_idx.find(target - nums[i])->second, i};
            }
            num_idx.emplace(nums[i], i);
        }
        return {-1, -1};
    }
};

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: list[int], target: int) -> int:
        return self.binary_search(nums, target)
    
    def binary_search(self, nums: list[int], target: int) -> int:
        def helper(l: int, r: int) -> int:
            print("l: " + str(l) + " r: " + str(r))
            if l > r:
                return -1
            mid: int = l + (r - l) // 2
            if target == nums[mid]:
                return mid
            elif target < nums[mid]:
                return helper(l, mid - 1)
            else:
                return helper(mid + 1, r)
        
        return helper(0, len(nums) - 1)

